### ISSUE TITLE
Forbid complex default values

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#427]
# Rule request

## Thesis

Today I saw this code:

```python
    def __init__(self, tree, filename='(none)', builtins=None,
                 withDoctest='PYFLAKES_DOCTEST' in os.environ, tokens=()):
```
https://github.com/PyCQA/pyflakes/blob/ac0aeba0eddb0b8e2285f5ac568e4621426b3aa7/pyflakes/checker.py#L635

What it does? It uses an expression as a default value. I am strongly against that.
We should forbid anything that is not a `ast.Name` or `ast.Attribute` to be used as a default value.

## Reasoning

It can be tricky. Nothing stops you from making database calls or http requests in such expressions.
It is also not readable for me. I would prefer to write:

```python
SHOULD_USE_DOCTEST = 'PYFLAKES_DOCTEST' in os.environ

def __init__(self, tree, filename='(none)', builtins=None,
                 withDoctest=SHOULD_USE_DOCTEST, tokens=()):
```

**Comment by kxepal** [comment]
At the same time are you fine with [`pytest.mark.skipIf`](https://docs.pytest.org/en/latest/skipping.html#id1) and [`pytest.mark.xfail`](https://docs.pytest.org/en/latest/skipping.html#reason-parameter) cases? How they are different from this one?

**Comment by mikhail-akimov** [comment]
I`ll do it!

**Comment by sobolevn** [comment]
@mikhail-akimov yes, you are absolutly correct.

You need to create:
1. a new `ast` violation in `best_practices.py`
2. then, create a new method in this class: https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/visitors/ast/functions.py#L98 call it `_check_argument_default_values`

**Comment by mikhail-akimov** [comment]
I found out that in order to prohibit the expression in default, it is enough to forbid `ast.Compare`, but what about `ast.Tuple`, `ast.NameConstant`, `ast.Tuple`? We have to forbid them too?

**Comment by mikhail-akimov** [comment]
What then will be the correct example?

**Comment by sobolevn** [comment]
The rule is: we should allow any immutable simple values and variables:
- `None`, `True`, `False`
- `1`, `2.0`
- `'some string'`, `b'some data string'`
- `(1, 2)`
- `variable_name`, `module.CONTANT`

**Comment by sobolevn** [comment]
Closed via https://github.com/wemake-services/wemake-python-styleguide/pull/486
